.travis.yml

```diff
@@ -11,22 +11,30 @@ matrix:
   allow_failures:
     - env: SHELL=zsh
 script:
-  - >
-    if ${LINT:-false}; then
-      shellcheck -s sh bin/shpec  # Just use th `sh` dialect
-    else
-      time $SHELL bin/shpec shpec/shpec_shpec.sh
-    fi
+  - |
+    ${LINT:-false} &&
+      exec shellcheck -s sh bin/shpec  # `sh` refers to POSIX
+    $SHELL bin/shpec shpec/shpec_shpec.sh
 cache: apt
 before_install:
-  - >
+  - |
+    get_shellcheck() {
+      (
+      for v 
```

Comment:
On 05/03/15 06:14, locochris wrote:
> In .travis.yml <https://github.com/rylnd/shpec/pull/79#discussion_r29552863>:
>
>>  cache: apt
>>  before_install:
>> -  - >
>> +  - |
>> +    get_shellcheck() {
>> +      (
>> +      for v
>
> @hlangeveld <https://github.com/hlangeveld> - gee really? - so you don't need to pass explicit arguments to |for|?

Yes.
There've been text in the man pages and the POSIX standard that it's only there
for backwards compatibility for as long as I can remember. That feature is not going anywhere.
But yes, I could make that explicit with "$@".

Functions should be small enough that you have no more than a single loop anyway ;-)

Here's my 'shpec/implicit_for_shpec.sh':
Note that _t() is the 'feature under test'

     describe "for loop without 'in'"
       _t() {
         (
         n=0
         for v
         do
           : $((n+=1))
         done
         echo $n
         )
       }
       it "skips empty argument lists"
         n=$(_t)
         assert equal $n 0
       end
       it "runs once for each argument"
         n=$(_t x y z)
         assert equal $n 3
       end
     end